src/HttpReply.h

```diff
@@ -161,5 +161,12 @@ class HttpReply: public Http::Message
     virtual bool parseFirstLine(const char *start, const char *end);
 };
 
+inline void
+HTTPMSGUNLOCK(HttpReply *&rep)
+{
+    if (Http::Message::Destroy(rep))
+        rep = nullptr;
```

Comment:
This nullification should be unconditional IMO. Same for the other two `HTTPMSGUNLOCK()` functions. Please see the review summary for details.

Reply:
Done.